### PR TITLE
prettierd: init at 0.23.3

### PR DESCRIPTION
**Pull request by n3oney**
###### Description of changes

#215416 seems abandoned, so I made a new PR, that works and uses mkYarnPackage instead of the node2nix script.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by JulienMalka** [comment]
@ofborg eval

**Review by JulienMalka — COMMENTED** [review]
I have tested this and it worked, so I'm not 100% sure why ofborg eval is failing (actually I am not sure why it is working on my machine), I think to package a node module like this you need to provide the hash to the deps, like explained in here : https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md

**Comment by n3oney** [comment]
> I think to package a node module like this you need to provide the hash to the deps, like explained in here : https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md

This is using mkYarnPackage, which doesn't have a hash option. I believe it works by just reading the yarn.lock file.
See eg. https://github.com/NixOS/nixpkgs/blob/d5d16930f3010c1ceeb85270c62cd3ccb00542b8/pkgs/tools/admin/synapse-admin/default.nix#L3

**Comment by JulienMalka** [comment]
> > I think to package a node module like this you need to provide the hash to the deps, like explained in here : https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md
> 
> This is using mkYarnPackage, which doesn't have a hash option. I believe it works by just reading the yarn.lock file. See eg.
> 
> https://github.com/NixOS/nixpkgs/blob/d5d16930f3010c1ceeb85270c62cd3ccb00542b8/pkgs/tools/admin/synapse-admin/default.nix#L3

Ah yes this looks right. With more sleep I now know what is wrong.

File: pkgs/development/tools/prettierd/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  '';
+
+  distPhase = "true";
+
+  meta = with lib; {
+    mainProgram = "prettier";
+    description = "prettier, as a daemon, for improved formatting speed.";
+    homepage = "https://github.com/fsouza/prettierd";
+    license = licenses.isc;
+    maintainers = with maintainers; [notashelf n3oney];
```
Comment (JulienMalka) [inline pkgs/development/tools/prettierd/default.nix:31]: Looks like notashelf does not exists

Comment (NotAShelf) [reply pkgs/development/tools/prettierd/default.nix:31]: I believe it is NotAShelf, I drafted the original derivation a little sloppy snd n3oney likely went with the one I drafted.

Comment (n3oney) [reply pkgs/development/tools/prettierd/default.nix]: whoops, sorry for that

**Comment by JulienMalka** [comment]
@ofborg eval

File: pkgs/development/tools/prettierd/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  mkYarnPackage,
+  fetchFromGitHub,
+}:
+mkYarnPackage rec {
+  pname = "prettierd";
+  version = "0.23.3";
+
+  src = fetchFromGitHub {
+    owner = "fsouza";
+    repo = pname;
```
Comment (drupol) [inline pkgs/development/tools/prettierd/default.nix:12]:
```suggestion
    repo = "prettierd";
```

File: pkgs/development/tools/prettierd/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    rev = "v${version}";
+    hash = "sha256-M9VA67Ix2aKS5V0cA0cFPXkASoAyFxW6rEopSYXtyiA=";
+  };
+
+  packageJSON = "${src}/package.json";
```
Comment (drupol) [inline pkgs/development/tools/prettierd/default.nix:17]:
```suggestion
  packageJSON = ./package.json;
```

File: pkgs/development/tools/prettierd/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  yarnLock = "${src}/yarn.lock";
```
Comment (drupol) [inline pkgs/development/tools/prettierd/default.nix:18]:
```suggestion
  yarnLock = ./yarn.lock;
```

File: pkgs/development/tools/prettierd/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  buildPhase = ''
+    yarn build
+  '';
```
Comment (drupol) [inline pkgs/development/tools/prettierd/default.nix:22]:
```suggestion
  buildPhase = ''
    runHook preBuild
    
    yarn build
    
    runHook postBuild
  '';
```

**Review by drupol — CHANGES_REQUESTED** [review]
Hey,

Thanks for your first pull request!

I made a review, let me know if you're feeling comfortable with it.

**Comment by drupol** [comment]
Something wrong here @n3oney ?

**Comment by n3oney** [comment]
> Something wrong here @n3oney ?

This PR is erroring and I was told (on Discord I believe) I need to package it some other way, but I don’t currently have the energy for that, so I decided to close the PR. Maybe @NotAShelf will pick up on the new implementation soon

**Comment by drupol** [comment]
Fair enough, thanks for the explanation!

**Pull request closed without merging**